packages/node/p2p/stream.go

```diff
@@ -210,8 +210,14 @@ func (m *Manager) handleStream(stream network.Stream) {
 	am := m.matchNewStream(stream)
 	if am != nil {
 		am.StreamChMutex.RLock()
+		defer am.StreamChMutex.RUnlock()
 		streamCh := am.StreamCh[protocolID]
-		am.StreamChMutex.RUnlock()
+
+		select {
+		case <-am.Ctx.Done():
+			return
+		default:
+		}
```

Comment:
Are we 100% sure this can't lead to a deadlock? because it is waiting on the channel while the `RLock` is still being held

Reply:
I was indeed adressing that =)